partcad/src/partcad/lint/schema.py

```
@@ -0,0 +1,50 @@
+from .lint import Linting, Severity, LintingReport
+
+
+class SchemaLinting(Linting):
+    def __init__(self, name: str, schema: dict) -> None:
+        super().__init__(name)
+        self.schema = schema
+
+    def get_targets(self, ctx: Context, package: Project) -> list[str]:
+        return [os.path.join(package.config_dir, "partcad.yaml")]
+
+    async def validate(self, ctx: Context, package: Project, target: str, lint_ctx: dict = {}) -> LintingReport:
```

Ruff (0.11.9)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)